library/Zend/Http/Client/Adapter/Curl.php

```diff
@@ -421,19 +421,26 @@ public function write($method, $uri, $httpVersion = 1.1, $headers = array(), $bo
             throw new AdapterException\RuntimeException("Error in cURL request: " . curl_error($this->curl));
         }
 
+        // separating header from body because it is dangerous to accidentially replace strings in the body
+        $responseHeaderSize = curl_getinfo($this->curl, CURLINFO_HEADER_SIZE);
+        $responseHeaders = substr($this->response, 0, $responseHeaderSize);
+
         // cURL automatically decodes chunked-messages, this means we have to disallow the Zend\Http\Response to do it again
```

Comment:
I separated header from body, to be sure that i don't manipulate the body. Possible test:
Textfile with Transfer-Encoding: chunked\r\n inside AND the webserver must not make use of chunking.
In the old version the assert that the file is the same would probably fail.